org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundSpgIterator.java

```diff
@@ -175,17 +122,17 @@ public void remove() {
 	public static Object[] getExcluders(Collection<?> regions) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)